### PR TITLE
[TASK] Add permalink tests and third party document permalink resolving

### PR DESCRIPTION
**Pull request by garvinhicking**
This does a couple of related things:

* Add unit testing for forwards (positive and negative expectations, version resolving, fallbacks)
* Refactor permalink resolving into a specific method that can be unit-tested
* Add unit test fixtures (objects.inv.json)
* Add script (spider.sh) to fetch unit test fixtures
* Refactor permalink resolving to deal with third party extension links (method resolveEntryPoint()
* Resolve "stable", "oldstable" and "dev" pseudo-version via composer package `t3docs/typo3-version-handling`
* Update `t3docs/typo3-version-handling` to latest version (to deal with 13.4 releases)

Happy reviewing :-)

### CODE REVIEW

File: legacy_hook/src/DocumentationLinker.php [hunk 1]
```diff
@@ -214,21 +228,42 @@ private function parseInventoryForIndex(string $index, array $json): string
...
+            // Core Version resolving. Uses the composer package t3docs/typo3-version-handling which allows to
+            // interpret strings as "dev", "stable", "oldstable" and can map "12" to latest 12.4.x version.
+            // If not resolvable, uses the raw version number as lookup (for example "12.4"). An invalid version
+            // string like "99.9999" will later fail when searching for the directory.
+            $resolvedVersionEnum = Typo3VersionMapping::tryFrom($version);
+            if ($resolvedVersionEnum === null) {
+                $resolvedVersion = $version;
+            } else {
+                $resolvedVersion = $resolvedVersionEnum->getVersion();
+            }
```
Comment (andreaskienast) [inline legacy_hook/src/DocumentationLinker.php:256]: Can be shortened to
```suggestion
            $resolvedVersion = Typo3VersionMapping::tryFrom($version)?->getVersion() ?? $version;
```
